### PR TITLE
Fix mismatched per-language option storages

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

File: src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs [hunk 1]
```diff
@@ -331,12 +339,12 @@ public bool TryFetch(LocalUserRegistryOptionPersister persister, OptionKey2 opti
...
         {"InlineRename_CollapseRenameUI", new RoamingProfileStorage("TextEditor.CollapseRenameUI")},
         {"InlineRename_UseInlineAdornment", new RoamingProfileStorage("TextEditor.RenameUseInlineAdornment")},
-        {"InlineRenameSessionOptions_PreviewChanges", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.PreviewRename")},
-        {"InlineRenameSessionOptions_RenameAsynchronously", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RenameAsynchronously")},
-        {"InlineRenameSessionOptions_RenameFile", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RenameFile")},
-        {"InlineRenameSessionOptions_RenameInComments", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RenameInComments")},
-        {"InlineRenameSessionOptions_RenameInStrings", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RenameInStrings")},
-        {"InlineRenameSessionOptions_RenameOverloads", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RenameOverloads")},
+        {"InlineRenameSessionOptions_PreviewChanges", new RoamingProfileStorage("TextEditor..Specific.PreviewRename")},
```
Comment (tmat) [inline src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs:342]: One time break for inline rename options.

File: src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs [hunk 2]
```diff
@@ -331,12 +339,12 @@ public bool TryFetch(LocalUserRegistryOptionPersister persister, OptionKey2 opti
...
         {"InlineRename_CollapseRenameUI", new RoamingProfileStorage("TextEditor.CollapseRenameUI")},
         {"InlineRename_UseInlineAdornment", new RoamingProfileStorage("TextEditor.RenameUseInlineAdornment")},
-        {"InlineRenameSessionOptions_PreviewChanges", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.PreviewRename")},
-        {"InlineRenameSessionOptions_RenameAsynchronously", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RenameAsynchronously")},
-        {"InlineRenameSessionOptions_RenameFile", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RenameFile")},
-        {"InlineRenameSessionOptions_RenameInComments", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RenameInComments")},
-        {"InlineRenameSessionOptions_RenameInStrings", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RenameInStrings")},
-        {"InlineRenameSessionOptions_RenameOverloads", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RenameOverloads")},
+        {"InlineRenameSessionOptions_PreviewChanges", new RoamingProfileStorage("TextEditor.Specific.PreviewRename")},
```
Comment (tmat) [inline src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs:342]: One time break of inline rename options.

File: src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs [hunk 3]
```diff
@@ -385,7 +393,7 @@ public bool TryFetch(LocalUserRegistryOptionPersister persister, OptionKey2 opti
         {"SolutionCrawlerOptionsStorage_BackgroundAnalysisScopeOption", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.BackgroundAnalysisScopeOption")},
         {"SolutionCrawlerOptionsStorage_CompilerDiagnosticsScopeOption", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.CompilerDiagnosticsScopeOption")},
         {"SplitCommentOptions_Enabled", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.SplitComments")},
-        {"SplitStringLiteralOptions_Enabled", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.SplitStringLiterals")},
+        {"SplitStringLiteralOptions_Enabled", new RoamingProfileStorage("TextEditor.CSharp.Specific.SplitStringLiterals")},
```
Comment (tmat) [inline src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs:396]: Another break. This option is only used by C# and declared as `Option2<T>`, so it does have `%LANGUAGE%` substituted to "".

File: src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs [hunk 4]
```diff
@@ -231,7 +239,7 @@ public bool TryFetch(LocalUserRegistryOptionPersister persister, OptionKey2 opti
 #pragma warning restore
         {"DocumentOutlineOptions_EnableDocumentOutline", new FeatureFlagStorage(@"Roslyn.DocumentOutline")},
         {"dotnet_code_quality_unused_parameters", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.UnusedParametersPreference")},
-        {"dotnet_remove_unnecessary_suppression_exclusions", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.RemoveUnnecessarySuppressionExclusions")},
+        {"dotnet_remove_unnecessary_suppression_exclusions", new RoamingProfileStorage("TextEditor..Specific.RemoveUnnecessarySuppressionExclusions")}, // Used to contain %LANGUAGE% but deosn't have per-language value, keeping the wrong name for compat.
```
Comment (Youssef1313) [inline src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs:242]: Is this exposed in Visual Studio UI options? If not, how can its value change other than from `.editorconfig`?

Comment (tmat) [reply src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs:242]: Good point. I don't see it.

Comment (tmat) [reply src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs:242]: @mavasani Should this option be configurable in VS UI?

Comment (tmat) [reply src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs]: Removed the storage for now, tracked by https://github.com/dotnet/roslyn/issues/66062

**Review by genlu — APPROVED** [review]
:shipit:

**Pull request merged**